tests/ml/test.ipynb

```diff
@@ -0,0 +1,247 @@
+{
```

Comment:
Could you please switch out this notebook for unit tests :)

Reply:
ok, i'll try it later in this month